frame/vesting/src/lib.rs

```diff
@@ -567,20 +567,29 @@ impl<T: Config> Pallet<T> {
 		(BalanceOf<T>, BoundedBTreeMap<T::VestingScheduleId, BalanceOf<T>, T::MaxVestingSchedules>),
 		DispatchError,
 	> {
-		let mut claims_per_schedule: BoundedBTreeMap<
-			T::VestingScheduleId,
-			BalanceOf<T>,
-			T::MaxVestingSchedules,
-		> = BoundedBTreeMap::new();
-
 		<VestingSchedules<T>>::try_mutate_exists(who, asset, |maybe_schedules| {
-			let mut total_balance_to_claim: BalanceOf<T> = Zero::zero();
 			match maybe_schedules {
-				Some(schedules) => {
-					vesting_schedule_ids
-						.into_all_ids(schedules)
-						.iter()
-						.map(|id_to_claim| {
+				Some(schedules) =>
+					vesting_schedule_ids.into_all_ids(schedules).iter().try_fold::<_, _, Result<
+						(
+							BalanceOf<T>,
+							BoundedBTreeMap<
+								T::VestingScheduleId,
+								BalanceOf<T>,
+								T::MaxVestingSchedules,
+							>,
+						),
```

Comment:
This type annotation is reduntant - type inference should be able to pick it up (see following suggestion)